src/python/WMCore/ACDC/DataCollectionService.py

```diff
@@ -20,26 +20,36 @@
 from WMCore.WMException import WMException
 
 
-def mergeFakeFiles(chunkFiles):
+def mergeSameFiles(chunkFiles):
     """
-    _mergeFakeFiles_
+    _mergeSameFiles_
 
     Receive a list of dicts with acdc files info and merge them together
-    (sum up events) belonging to the same MCFakeFile.
+    (sum up events) belonging to the same File.
     """
-    # do not do anything for real files
-    if not chunkFiles[0]['lfn'].startswith('MCFakeFile'):
-        return chunkFiles
-
-    logging.info("Merging %d ACDC FakeFiles...", len(chunkFiles))
+    originalLen = len(chunkFiles)
+    logging.info("Merging %d ACDC files with the same lfn...", originalLen)
     mergedFiles = {}
     for acdcFile in chunkFiles:
         if acdcFile['lfn'] not in mergedFiles:
             mergedFiles[acdcFile['lfn']] = acdcFile
         else:
-            mergedFiles[acdcFile['lfn']]['events'] += acdcFile['events']
-            mergedFiles[acdcFile['lfn']]['runs'][0]['lumis'].extend(acdcFile['runs'][0]['lumis'])
-    logging.info("resulted in %d final ACDC FakeFiles.", len(mergedFiles))
+            # if it is fake file events add up for real file don't add up.
+            if chunkFiles[0]['lfn'].startswith('MCFakeFile'):
+                mergedFiles[acdcFile['lfn']]['events'] += acdcFile['events']
+
+            for runLumi in acdcFile['runs']:
+                containRunNum = False
+                for mRunLumi in mergedFiles[acdcFile['lfn']]['runs']:
```

Comment:
I think `runs` is a set here, right? Can't we simply add new Run objects to this set instead of trying to merge them?

Reply:
Actually here is the list of dict 
it wrapped run object later (below code)

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ACDC/DataCollectionService.py#L260